### PR TITLE
Setting up Github Actions & Dependabot

### LINKED ISSUE #68: [Enhancement] Github Status Check
## Enhancement

Since our project is at a reasonable size, we would like to integrate Github actions to improve code delivery.

## Specifications

We would like you to set up [Github workflows](https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs) to automate actions on PRs. Ideally, we want to at least enable automated unit testings. You can also explore other Actions on Github marketplace, such as [automated Prettier](https://github.com/creyD/prettier_action).

This is an open-ended task, so be prepared for some self-learning and exploration! As always, feel free to ping us to discuss about any problems!

### PR DESCRIPTION
**Pull request by JuliaZel**
# Description

Added Github actions for automatic unit testing and stylistic reformatting of code. 

For the Prettier action, note that it will run when a pull request is made to main. It currently is running on all files (not just changed files) and will go and fix ant formatting issues. For the future, this could be changed with the "only_changed" option once all users have finished their current tickets and have the workflows in their directories (re-pull from main). 

For Dependabot, we have enabled security alerts, which will have it automatically open PRs with security issues and it should starting now also add @nickbar01234 as a reviewer automatically. 

Unit Testing will run on every commit! 

Thank you! 

## Issues
Resolve #68 

## Test

Created many dummy branches and PRs to our own branch in order to test that it will run on PR. 
image

## Possible Downsides

Note that the prettier action can take a little bit of time to fully run. Will become faster once we only run it on changed files.

### CODE REVIEW

File: .github/workflows/prettier.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          node-version: "18.x"
+
+      - name: NPM Install
+        run: npm i
+
+      - name: Prettify code
+        uses: creyD/prettier_action@v4.3
+        with:
+          prettier_options: "--write **/*.tsx"
```
Comment (nickbar01234) [inline .github/workflows/prettier.yml:27]: We also have `mjs` and `ts` files in the repository. Is there a glob pattern for all 3 types of files?

File: src/app/api/file/route.client.ts [hunk 2]
```diff
@@ -11,6 +11,11 @@ type IFile = z.infer<typeof File>;
  * Extends the parameters of fetch() function to give types to the RequestBody.
  */
 interface IRequest extends Omit<RequestInit, "body"> {
+
```
Comment (nickbar01234) [inline src/app/api/file/route.client.ts:14]: How come we added a bunch of space here?

File: src/app/private/[uid]/user/layout.tsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 "use client";
 
+import Sidebar, { ISideBar } from "@components/Sidebar";
```
Comment (nickbar01234) [inline src/app/private/[uid]/user/layout.tsx:3]: I don't think this import statement is necessary, since it's not used.

**Review by nickbar01234 — APPROVED** [review]
The Github actions looks amazing, thanks! There's 2 small comments that needs addressing, and you guys can merge once that's done! 

P.S: Remember to squash the branch when you merge.

File: src/app/private/[uid]/user/layout.tsx (hunk 3 above)
Comment (sburchfield33) [reply src/app/private/[uid]/user/layout.tsx:3]: I think the import is used though, note lines 14, 38 in the file.

**Pull request merged**